core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1190,8 +1192,31 @@ class KafkaApis(val requestChannel: RequestChannel,
     val metadataRequest = request.body[MetadataRequest]
     val requestVersion = request.header.apiVersion
 
+    // Check if topicId is presented firstly.
+    val topicIds = metadataRequest.topicIds.asScala.toSet.filterNot(_ == Uuid.ZERO_UUID)
+    val supportedVersionTopicIds = if (config.interBrokerProtocolVersion >= KAFKA_2_8_IV1) topicIds else Set.empty[Uuid]
+
+    val unsupportedVersionTopicIds = topicIds.diff(supportedVersionTopicIds)
```

Comment:
I'm curious if this diff is needed. It seems that the above check will result in all topicIds being left out or all of them being included.